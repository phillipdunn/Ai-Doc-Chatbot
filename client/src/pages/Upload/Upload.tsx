import { Box, Button, Typography } from '@mui/material';
import React, { FC } from 'react';
import CloudUploadIcon from '@mui/icons-material/CloudUpload';
import { styled } from '@mui/material/styles';

interface Props { }

const VisuallyHiddenInput = styled('input')({
  clip: 'rect(0 0 0 0)',
  clipPath: 'inset(50%)',
  height: 1,
  overflow: 'hidden',
  position: 'absolute',
  bottom: 0,
  left: 0,
  whiteSpace: 'nowrap',
  width: 1,
});

const Upload: FC<Props> = () => {

  const handleSubmit = (e: any) => {
    console.log(e.target.files[0]);
  };

  return (
    <Box sx={{ p: 2 }}>
      <Typography variant="h4" sx={{ mt: 4 }}>Upload a file</Typography>

      <Button
        sx={{ mt: 2 }}
        component="label"
        role='button'
        variant="contained"
        tabIndex={-1}
        startIcon={<CloudUploadIcon />}
      >
        Upload file
        <VisuallyHiddenInput type="file" onChange={handleSubmit} />
      </Button>
    </Box>
  );
};

export default Upload;
