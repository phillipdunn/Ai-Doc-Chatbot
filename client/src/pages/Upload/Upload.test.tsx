import React from 'react';
import { create } from 'react-test-renderer'; // Import the 'create' function
import Upload from './Upload';

describe('Upload', () => {
    it('should render without crashing', () => {
        const component = create(<Upload />);
        expect(component).toBeTruthy();
    });
});
