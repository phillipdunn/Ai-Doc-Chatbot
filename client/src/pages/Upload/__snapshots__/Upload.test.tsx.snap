// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload renders 1`] = `
<div>
  Upload renders
</div>
`;
